### ISSUE TITLE
False positive for RedundantCondition after update from 3.10.1 to 3.11.0

### ISSUE THREAD

**Issue by kubawerlos** [vimeo/psalm#3125]
After [update](https://github.com/kubawerlos/types-checker/pull/26) from 3.10.1 to 3.11.0 one false positive started to being reported, For a code;
```php
<?php
class Foo
{
    public function test() : void
    {
        $x = $this->get();
        $i = 0;
        while (isset($x[$i]) && is_array($x[$i])) {
            echo $i;
            $i++;
        }
    }
    
    /**
     * @return array
     */
    private function get() : array
    {
        return [['a'], 'b', 'c', ['d']];
    }
}
```

The function `get` returns array that has mixed elements - string or an array, so `is_array` is needed.

https://psalm.dev/r/48ff1bea43

**Comment by kubawerlos** [comment]
Thank you @muglug
